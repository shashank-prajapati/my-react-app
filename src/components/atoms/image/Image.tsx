import React from "react";

type Props = {
  source: string;
  imgWidth?: string | number;
  imgHeight?: string | number;
};

const Image = (props: Props) => {
  const { source, imgHeight, imgWidth } = props;
  return (
    <img
      style={{ display: "block", width:imgWidth ?? "inherit" , height:imgHeight ?? "inherit" }}
      src={source}
      //height={height ?? "inherit"}
      alt="Book"
      //width={width ?? "inherit"}
    ></img>
  );
};

export default Image;
