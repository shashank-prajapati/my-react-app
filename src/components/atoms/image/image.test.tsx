import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Image from './Image';
import coverImage from "../../../media/coverImage.png";


describe('Testing ImageAtom', ()=>{
    it('should match the default snapshot',()=>{
        const {container} = render(<Image source={coverImage}/>,);
        expect(container).toMatchSnapshot();
    })
})

