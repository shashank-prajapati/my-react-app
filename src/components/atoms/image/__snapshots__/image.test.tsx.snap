// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ImageAtom should match the default snapshot 1`] = `
<div>
  <img
    alt="Book"
    height="inherit"
    src="coverImage.png"
    style="display: block;"
    width="inherit"
  />
</div>
`;
